src/server/engine_shard_set.cc

```diff
@@ -70,6 +79,66 @@ ShardMemUsage ReadShardMemUsage(float wasted_ratio) {
   return usage;
 }
 
+// TODO: document this class.
+// Thread safe.
+class RoundRobinSharder {
+ public:
+  void InitThreadLocal() {
+    round_robin_prefix_ = absl::GetFlag(FLAGS_round_robin_prefix);
+
+    if (IsEnabled()) {
+      // 500k entries will consume 1mb per thread, and will allow 100 keys with < 1% collision
+      // probability. Since this has a considerable footprint, we only allocate when enabled.
+      constexpr size_t kRoundRobinSize = 500'000;
+      round_robin_shards_.resize(kRoundRobinSize);
```

Comment:
yes but then all the code becomes more complicated, this is just a lookup table

Reply:
in addition to added complexity, it will also add allocations on the hot path.
Resizing once is more efficient, and here it's done during initialization